sdk/servicebus/service-bus/review/service-bus.api.md

```diff
@@ -287,12 +328,12 @@ export class ServiceBusManagementClient extends ServiceClient {
     constructor(connectionString: string, options?: ServiceBusManagementClientOptions);
     constructor(fullyQualifiedNamespace: string, credential: TokenCredential, options?: ServiceBusManagementClientOptions);
     createQueue(queueName: string, operationOptions?: OperationOptions): Promise<QueueResponse>;
-    createQueue(queue: QueueProperties, operationOptions?: OperationOptions): Promise<QueueResponse>;
+    createQueue(queue: CreateQueueOptions, operationOptions?: OperationOptions): Promise<QueueResponse>;
     createRule(topicName: string, subscriptionName: string, rule: RuleProperties, operationOptions?: OperationOptions): Promise<RuleResponse>;
```

Comment:
Does this concept of "options" being passed to create methods and properties being passed to update methods not apply to rules?

Reply:
Currently, the way `update` works is that it overrides the "rule-1" if the "rule-1" already exists.
This does feel like what a user would expect. I'm discussing this offline with Josh too.
Asking users to do `getRule` before `updateRule` feels unnecessary. We can further discuss this.

Nevertheless, I'll update the RuleProps similar to the rest of the update methods for consistency.

Reply:
In that case, I would not recommend treating rule similar to others.
Use `RuleProperties` for create as well as update methods

Reply:
We are using CreateRuleOptions in .NET for consistency.

Reply:
I kinda convinced myself to have a separate `CreateRuleOptions`.

**Scenario**
Say there is a rule with `{name: "rule-1", filter: <filter-1>, action: <action-1>}`
User can unknowingly override the rule with `{name: "rule-1", filter: <filter-2>}` with the update API and can miss retaining the  `action`. 
This would make the action to be `undefined` after the update operation.
(Basically, I'm extrapolating the same problem we have with the other update operations.)

So, maybe doing a get request does make sense.

Reply:
I understand, but now this deviates from our policy of having everything in an *Options interface as optional.
Also, `filter` should be mandatory when creating the rule.
So, I propose the below instead of having a `CreateRuleOptions`

```js
createRule(topicName: string, subscriptionName: string, ruleName: string, ruleFilter: SqlRuleFilter | CorrelationRuleFilter, operationOptions?: OperationOptions): Promise<RuleResponse>;
createRule(topicName: string, subscriptionName: string, ruleName: string, ruleFilter: SqlRuleFilter | CorrelationRuleFilter, ruleAction: SqlRuleAction, operationOptions?: OperationOptions): Promise<RuleResponse>;
```

Reply:
Users can create a placeholder rule without providing the ruleFilter, which would then get the default rule.
This won't be allowed with your proposal.

Reply:
How about?
```ts
// CreateRuleOptions extends OperationOptions
createRule(topicName: string, subscriptionName: string, ruleName: string, options?: CreateRuleOptions): Promise<RuleResponse>;

interface RuleProperties {
    action: SqlRuleAction;
    filter: SqlRuleFilter | CorrelationRuleFilter;
}
updateRule(topicName: string, subscriptionName: string, ruleName: string, options: RuleProperties & OperationOptions): Promise<RuleResponse>;
```

Reply:
I thought of this approach first as well to mirror what we do with queues and the others.
But, the problem here is that filter is the main part of the rule. It is not an afterthought like it is for the queue options.
Most customers wanting to create a rule will want to provide a filter.

The case of the default rule that you mentioned can be achieved by customer providing a true filter.
Or we can expose a constant for the default true filter value which can be used by the customer.
Trying to optimize for this case, is optimizing for corner cases rather than most commonly used cases.

Reply:
```ts
interface RuleProperties {
    name: string;
    action: SqlRuleAction;
    filter: SqlRuleFilter | CorrelationRuleFilter;
}

interface RuleResponse extends RuleProperties, Response {}

createRule(topicName: string, subscriptionName: string, ruleName: string, ruleFilter: SqlRuleFilter | CorrelationRuleFilter, operationOptions?: OperationOptions): Promise<RuleResponse>;
createRule(topicName: string, subscriptionName: string, ruleName: string, ruleFilter: SqlRuleFilter | CorrelationRuleFilter, ruleAction: SqlRuleAction, operationOptions?: OperationOptions): Promise<RuleResponse>;

// DOCS: Call getRule, use the response in the updateRule
updateRule(topicName: string, subscriptionName: string, rule: RuleProperties, operationOptions?: OperationOptions): Promise<RuleResponse>;
```
This is your suggestion, right?

Reply:
yes

Reply:
Updated